### ISSUE TITLE
[typedef] is also linting .js files

### ISSUE THREAD

**Issue by GarfieldDeepClone** [typescript-eslint/typescript-eslint#1836]
**Repro**

The typedef rule is linting also js files and therefore I have a lot of errors. I have others @typescript-eslint rules active which do not causes such problems.

```JSON
{
"parser": "@typescript-eslint/parser",
	"parserOptions": {
		"ecmaVersion": 6,
		"ecmaFeatures": {
			"blockBindings": false,
			"forOf": false
		}
	},
"plugins": [
		"@typescript-eslint"
	],
  "rules": {
		"@typescript-eslint/typedef": [
			"error", {
				"parameter": true,
				"propertyDeclaration": true,
				"arrowParameter": false,
				"memberVariableDeclaration": false,
				"arrayDestructuring": false,
				"objectDestructuring": false,
				"variableDeclaration": false,
				"variableDeclarationIgnoreFunction": false
			}
		],
}
```
myFileThatShouldNotBeLinted.js:
```JS
angular
	.module('myModule')

	.controller('MyController', function(
		$uiRouterGlobals,
		xyService,
		zService,
	)
...
```

**Expected Result**
$uiRouterGlobals, xyService, zService shouldn't be linted by this rule, because it's in a js file.
Not linting *.js files
**Actual Result**
It's linting .js files
**Additional Info**
I can see the error in IntelliJ and when running eslint via grunt task.

Did eslint throw an exception?
no

Please run your lint again with the --debug flag, and dump the output below.
i.e. eslint --ext ".ts,.js" src --debug

eslint:file-enumerator Enter the directory: x/y/z +0ms
  eslint:cascading-config-array-factory Load config files for x/y/z. +0ms
  eslint:cascading-config-array-factory No cache found: x/y/z. +0ms
  eslint:config-array-factory Config file not found on x/y/z +0ms
  eslint:cascading-config-array-factory Cache hit: x/y. +0ms
  eslint:ignore-pattern Check { filePath:
   'x/y/z/myFileThatShouldNotBeLinted.js',
  dot: false,
  relativePath: 'z/myFileThatShouldNotBeLinted.js',
  result: false } +0ms
  eslint:file-enumerator Yield: myFileThatShouldNotBeLinted.js +0ms
  eslint:cascading-config-array-factory Load config files for x/y/z. +0ms
  eslint:cascading-config-array-factory Cache hit: x/y/z. +0ms
  eslint:cli-engine Lint x/y/z/myFileThatShouldNotBeLinted.js +0ms
  eslint:linter Linting code for x/y/z/myFileThatShouldNotBeLinted.js (pass 1) +0ms
  eslint:linter Verify +1ms
  eslint:linter With ConfigArray: x/y/z/myFileThatShouldNotBeLinted.js +0ms
  eslint:linter Generating fixed text for x/y/z/myFileThatShouldNotBeLinted.js (pass 1) +117ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +1ms
  eslint:ignore-pattern Check { filePath:
   'x/y/z/myFileThatShouldNotBeLinted.js',
  dot: false,
  relativePath: 'z/myFileThatShouldNotBeLinted.js',
  result: false } +122ms

 10:3   error  expected $uiRouterGlobals to have a type annotation        @typescript-eslint/typedef

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.25.0` |
| `@typescript-eslint/parser`        | `2.25.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `6.3.0` |
| `node`                             | `10.16.3` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
This is intended. The rules do not check file extensions for a number of reasons.

If you would like to use the rule in a mixed js/ts codebase, you should use eslint's overrides config to turn off rules that aren't compatible with js files:

https://eslint.org/docs/user-guide/configuring#disabling-rules-only-for-a-group-of-files
